Authentication/SignUp/View/SignupController.swift

```diff
@@ -65,62 +67,120 @@ private extension SignupController {
         bindPasswordElements()
         bindButtons()
         setTextfieldOrder()
+        
+        _viewModel.signUpExecuting.observeNext { [unowned self] executing in
+            if executing {
+                self._delegate.signupControllerWillExecuteSignUp(self)
+            } else {
+                self._delegate.signupControllerDidExecuteSignUp(self)
+            }
+            self.signupView.signUpButtonPressed = executing
+        }
+        
+        _viewModel.signUpErrors.observeNext { [unowned self] in self._delegate.signupController(self, didSignUpWithError: $0) }
     }
     
     private func bindNameElements() {
         if let nameTextField = signupView.usernameTextField {
             _viewModel.name <~ nameTextField.rex_textSignal
             _viewModel.nameValidationErrors.signal.observeNext { [unowned self] errors in
-                self._delegate.signupController(self, didFailNameValidationWithErrors: errors)
+                if errors.isEmpty {
+                    self._delegate.signupControllerDidPassNameValidation(self)
```

Comment:
I think we should change this long naming (brought from objc conventions) and have it written as `didPassNameValidation(self)`